boskos/ranch/ranch_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ranch
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/test-infra/boskos/common"
+)
+
+func MakeFakeClient(resources []common.Resource) *Ranch {
+	newRanch := &Ranch{
+		Resources: resources,
+	}
+
+	return newRanch
+}
+
+func TestSyncConfig(t *testing.T) {
+	var testcases = []struct {
```

Comment:
Consider adding more complex scenarios. For example, adding/removing multiple, appending when oldRes is not empty, delete where some are busy and others are free.

Also: multiple types with the same name